gnssanalysis/gn_download.py

```diff
@@ -283,7 +262,62 @@ def check_file_present(comp_filename: str, dwndir: str) -> bool:
     return present
 
 
-def check_n_download_url(url, dwndir, filename=False, uncomp=False):
+def decompress_file(input_filepath: _Path, delete_after_decompression: bool = False) -> _Path:
+    """
+    Given the file path to a compressed file, decompress it in-place
+    Assumption is that filename of final file is the stem of the compressed filename for .gz files
+    Option to delete original compressed file after decompression (Default: False)
+    """
+    # Get absolulte path
+    input_file = input_filepath.resolve()
```

Comment:
None of the functions need a fully resolved path, I'm just making sure that the `output_file` is clearly defined when we return it to whatever function is calling it in future (not sure it will be necessary but in case there is some corner case that needs it)